queryFields.js

```diff
@@ -0,0 +1,139 @@
+// @ts-check
```

Comment:
What is this?

Reply:
I do write a lot of typescript and I'm used to work with the preprocessor for types. The line enables typescript preprocessor so I can add types and type checks. This is really useful for rapid dev on GraphQL types especially as they are very complex.

More info: https://code.visualstudio.com/docs/languages/javascript#_type-checking

In this case, types were used only for development time and they will be not exposed to end users - methods are mounted to the context. I was planning to remove this after review.